objs/egg.cc

```diff
@@ -858,6 +1030,109 @@ bool Egg_object::is_active(
 	}
 }
 
+// test_unhatch pretty much inverts all the criteria compared to is_active
+// thiswas added for improve jukebox_egg behaviour to give the eggs a proximity
+// near and far trigger unless other egg types are modified this code should not
+// effect anything except jukebox_eggs
+
+bool Egg_object::test_unhatch(
```

Comment:
I wonder if the logic we have in weather effects for weather eggs should not use this instead?